src/components/Gallery.svelte

```diff
@@ -12,19 +11,16 @@
         dispatch('showDetail', { movie, position });
     }
 
-    function getFlyParams(node: HTMLElement) {
-        const rect = node.getBoundingClientRect();
-        return {
-            x: rect.left + rect.width / 2 - window.innerWidth / 2,
-            y: rect.top + rect.height / 2 - window.innerHeight / 2,
-            duration: 500
-        };
+    function handleKeydown(event: KeyboardEvent, movie: Movie) {
+        if (event.key === 'Enter' || event.code === 'Space') {
+            showDetail(movie, event as unknown as MouseEvent);
```

Comment:
Instead of casting this event (which is potentially very bug prone), you could also make the `showDetail` accept an event target, which is in both the KeyboardEvent as the MouseEvent.

Reply:
I understand what you mean this should solve it! :

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-bfe0e412488effea14eec1aad806e9297fb0b24e3c35f7605aae1c76597d4ffd